pyspecdata/core.py

```diff
@@ -7282,32 +7237,33 @@ def fit(self,set = None, set_to = None, force_analytical = False, silent = False
             leastsq_kwargs.update({'Dfun':self.parameter_gradient})
```

Comment:
so, my understanding is that this doesn't work right now? (We are not generating Dfun correctly?)

I would recommend to make sure that you have your most up-to-date version of your attempt at Dfun using sympy diff saved in a branch that's different from `alec_fitdata` (typically, just check out the correct commit, and do `git checkout -b`, or even easier right-click on the correct commit in gitk and do "create branch"), and then please generate a new pull request using that branch.

Reply:
Yes, Dfun does not work. I tried a few things quite a while ago and did not have success.

Reply:
I will checkout a new branch then., it is likely stored in the git tree of this branch. I will check now.

Reply:
Ok - after going through the git tree, I found what was my stopping point for debugging Dfun and have checked that out as its own branch. Does this need to be pushed?

Reply:
yes, please push that, and then set it up as a pull request

Reply:
Ok, it has been pushed + set up as pull request

Reply:
To me there are 4 lines of code that are obvious for me to delete (which I am putting below) -- but these call things like parameter_gradient, so should that be deleted too?
`        if hasattr(self,'has_grad') and self.has_grad == True:
            leastsq_kwargs.update({'Dfun':self.parameter_gradient})
        if 'Dfun' in list(leastsq_kwargs.keys()):
            if not silent: print("yes, Dfun passed with arg",leastsq_kwargs['Dfun'])
`

Reply:
I think so.

Remember that you saved a branch where you are actually using `diff` to calculate the parameter gradients, and all this stuff (should?) still be there.

Reply:
Yes, that is correct. so I will delete this bit of code and the associated.